Add filtering/exclude of rows in SummaryTable

jorgenherje opened the pull request:
Closes: #10139

magnesj commented:
I get some invalid signal/slot connections

![image](https://user-images.githubusercontent.com/1793152/235147874-f19e0d8f-f443-4a80-b372-efeaab069efa.png)

magnesj reviewed (changes requested):
Some signal/slots are connected to a null object.
When opening from a project file, the summary table is empty

jorgenherje commented:
> Some signal/slots are connected to a null object. When opening from a project file, the summary table is empty

- Added guard for nullptr of `plotDefinition` in `RiuQwtPlotWidget`. Previous assert has been removed and usage of `plotDefinition` guards has been utilized - however guard when connecting signals/slots was forgotten then. 
- Fixed incorrect loading of data resulting in emtpy tabled when loading project.
